routes/api/v1/favorites.js

```diff
@@ -4,6 +4,67 @@ var User = require('../../../models').User;
 var Location = require('../../../models').Location;
 var Favorite = require('../../../models').Favorite;
 const titleize = require('titleize');
+const fetch = require('node-fetch');
+
+var getWeather = function(location) {
+  fetch(`https://api.darksky.net/forecast/${process.env.DARKSKY_KEY}/${location.latitude},${location.longitude}`)
+  .then(response => response.json())
+  .then(function(data) {
+    // data now returns the hash from DarkSky that I want
+    var pry = require('pryjs'); eval(pry.it)
+    return data // this was my attempt to try to get data out of the helper method
+  })
```

Comment:
In my experience so far the trick is that you're getting "Promises all the way down." So you need to do returns all the way back up. For example, my getWeather looks like this.
```
var get_weather = function(location){
  return axios.get(`https://api.darksky.net/forecast/${process.env.DARK_SKY_API_KEY}/${location}`)
  .then(response => {
    return response.data;
  })
```
Notice the return on the response.data *and* on the axios.get

Additionally, even when you get the return all the way to the top like that, when you call it in your route, it will still return a promise. You need another .then to retrieve the value, like so.

```
if (req.body.api_key){
    verifyApiKey(req.body.api_key)
    .then((validity) => {
      if (validity == false) {
        res.setHeader("Content-Type", "application/json");
        res.status(401).send("No user with that key!")
      } else {
        get_location(req.query.location)
        .then(location => {
          var parsed_location = `${location["lat"]},${location["lng"]}`
          return get_weather(parsed_location)
          .then(weather => {
            final_forecast = parseWeather(req.query.location, weather)
            res.setHeader("Content-Type", "application/json");
            res.status(200).send(JSON.stringify(final_forecast));
          })
          .catch(error => {
            return error;
          });
```
Every single helper I wrote here still needed a .then in the router to retrieve and work with the returns I expected.

Hope this helps!

Reply:
Thanks Matt, I will give that a shot and see what I can find.